patch_api/api/orders_api.py

```diff
@@ -666,10 +705,29 @@ def retrieve_orders_with_http_info(self, **kwargs):  # noqa: E501
         path_params = {}
 
         query_params = []
-        for key in kwargs:
-            query_params.append([key, kwargs.get(key)])
         if "page" in local_var_params:
             query_params.append(("page", local_var_params["page"]))  # noqa: E501
+        if "metadata" in local_var_params:
+            query_params.append(
+                ("metadata", local_var_params["metadata"])
+            )  # noqa: E501
+        if "metadata_ext_id" in local_var_params:
+            query_params.append(
+                ("metadata[ext_id]", local_var_params["metadata_ext_id"])
+            )  # noqa: E501
+        if "metadata_disregarded" in local_var_params:
+            query_params.append(
+                ("metadata[disregarded]", local_var_params["metadata_disregarded"])
+            )  # noqa: E501
+
+        # do not add duplicate keys to query_params list
+        existing_keys = []
+        for param in query_params:
+            existing_keys.append(param[0])
+
+        for key in kwargs:
+            if key not in existing_keys:
+                query_params.append([key, kwargs.get(key)])
```

Comment:
this is the place in which this new logic is most relevant. Previously, a named query param would appear twice because all args were added to the params list even if they already matched one of the conditionals above